logstash-core/lib/logstash/output_delegator_strategy_registry.rb

```diff
@@ -0,0 +1,28 @@
+module LogStash; class OutputDelegatorStrategyRegistry
+  # This is generally used as a singleton
+  # Except perhaps during testing
+  def self.instance
+    @instance ||= self.new
```

Comment:
I'm not using singleton here because during testing I do want the ability to create a new instance without the default plugin list. I think we need a better name for this pattern (do you know of one that is already in use?) as `instance` too strongly implies a singleton. 

This is more like 'it works as a singleton by default but it is totally legit to make another instance, which is very useful for testing.'